src/Aspire.Hosting/Publishing/ManifestPublisher.cs

```diff
@@ -132,13 +132,26 @@ private static void WriteExecutable(ExecutableResource executable, ManifestPubli
         context.Writer.WriteString("workingDirectory", relativePathToProjectFile);
 
         context.Writer.WriteString("command", executable.Command);
-        context.Writer.WriteStartArray("args");
 
-        foreach (var arg in executable.Args ?? [])
+        var args = new List<string>(executable.Args ?? []);
+        if (executable.TryGetAnnotationsOfType<ExecutableArgsCallbackAnnotation>(out var argsCallback))
         {
-            context.Writer.WriteStringValue(arg);
+            foreach (var callback in argsCallback)
+            {
+                callback.Callback(args);
+            }
+        }
+
+        if (args.Count > 0)
```

Comment:
Previously if `ExecutableResource.Args` was empty the "args" would be in the manifest with an empty array. I changed this to not include the `args` to be consistence with Container manifest.

Reply:
Add a test to verify this?

Reply:
Done.